pm/recipient_test.go

```diff
@@ -0,0 +1,436 @@
+		t.Error("expected store recipientRand to match ticket recipientRandHash")
+	}
+
+	// Test invalid recipientRand revealed
+	// Config stub broker with zero deposit and non-zero penalty escrow
+	b.SetDeposit(sender, big.NewInt(0))
+	b.SetPenaltyEscrow(sender, big.NewInt(500))
+	// Redeem ticket to invalidate recipientRand
+	if err := r.RedeemWinningTicket(ticket.Hash()); err != nil {
+		t.Fatal(err)
+	}
+
+	// New ticket with same invalid recipientRand, but updated senderNonce
+	ticket = &Ticket{
+		Recipient:         ethcommon.Address{},
+		Sender:            sender,
+		FaceValue:         faceValue,
+		WinProb:           winProb,
+		SenderNonce:       1,
+		RecipientRandHash: params.RecipientRandHash,
+	}
+
+	_, err = r.ReceiveTicket(ticket, sig, params.Seed)
+	if err == nil {
+		t.Error("expected invalid recipientRand revealed error")
+	}
+	if err != nil && err != errInvalidRecipientRandRevealed {
+		t.Errorf("expected invalid recipientRand revealed error, got %v", err)
+	}
+
+	// Test invalid senderNonce
+	params, err = r.TicketParams(sender)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Receive senderNonce = 0
+	ticket0 := &Ticket{
+		Recipient:         ethcommon.Address{},
+		Sender:            sender,
+		FaceValue:         faceValue,
+		WinProb:           winProb,
+		SenderNonce:       0,
+		RecipientRandHash: params.RecipientRandHash,
+	}
+
+	_, err = r.ReceiveTicket(ticket0, sig, params.Seed)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Receive senderNonce = 1
+	ticket1 := &Ticket{
+		Recipient:         ethcommon.Address{},
+		Sender:            sender,
+		FaceValue:         faceValue,
+		WinProb:           winProb,
+		SenderNonce:       1,
+		RecipientRandHash: params.RecipientRandHash,
+	}
+
+	_, err = r.ReceiveTicket(ticket1, sig, params.Seed)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Replay senderNonce = 1 (new nonce = highest seen nonce)
+	_, err = r.ReceiveTicket(ticket1, sig, params.Seed)
+	if err == nil {
+		t.Error("expected invalid senderNonce (new nonce = highest seen nonce) error")
+	}
+	if err != nil && err != errInvalidTicketSenderNonce {
+		t.Errorf("expected invalid senderNonce (new nonce = highest seen nonce) error, got %v", err)
+	}
+
+	// Replay senderNonce = 0 (new nonce < highest seen nonce)
+	_, err = r.ReceiveTicket(ticket0, sig, params.Seed)
+	if err == nil {
+		t.Error("expected invalid senderNonce (new nonce < highest seen nonce) error")
+	}
+	if err != nil && err != errInvalidTicketSenderNonce {
+		t.Errorf("expected invalid senderNonce (new nonce < highest seen nonce) error, got %v", err)
+	}
+}
+
+func TestRedeemWinningTicket(t *testing.T) {
+	sender := ethcommon.HexToAddress("A69cdA26600c155cF2c150964Bdb5371ac3f606F")
+	b := newStubBroker()
+	v := &stubValidator{}
+	ts := newStubTicketStore()
+	secret := [32]byte{3}
+	faceValue := big.NewInt(100)
+	winProb := big.NewInt(100)
+	sig := []byte("foo")
+
+	r := NewRecipient(b, v, ts, secret, faceValue, winProb)
```

Comment:
Perhaps extract this line and the lines above it into a helper function that creates a recipient with some default setup? ([example here](https://github.com/livepeer/go-livepeer/pull/651/files#diff-0800fac1c4a865f1fd74c8aeebcd8e60R199)).